protocol/bc/entry.go

```diff
@@ -118,7 +118,13 @@ func writeForHash(w io.Writer, c interface{}) error {
 		return nil
 
 	case reflect.Struct:
-		for i := 0; i < v.NumField(); i++ {
+		typ := v.Type()
+		for i := 0; i < typ.NumField(); i++ {
+			sf := typ.Field(i)
+			if sf.Tag.Get("entry") == "-" {
```

Comment:
Seems perfect to me. Tags aren't so pervasive that I would be worried about [clashing with something else](https://github.com/golang/go/wiki/Well-known-struct-tags), and it's perfectly clear in context.